openmdao/solvers/linesearch/tests/test_backtracking.py

```diff
@@ -254,12 +254,60 @@ def linearize(self, inputs, outputs, jac):
         jac[('z', 'z')] = -inputs['x'] + 1.0
         jac[('z', 'x')] = -outputs['z']
 
+class ImplCompTwoStatesAEBug(om.ImplicitComponent):
+
+    def setup(self):
+        self.add_input('x', 0.5)
+        self.add_output('y', 0.0)
+        self.add_output('z', 2.0, lower=1.5, upper=2.5)
+
+        self.maxiter = 10
+        self.atol = 1.0e-12
+
+        self.declare_partials(of='*', wrt='*')
+
+        self.counter = 0
+
+    def apply_nonlinear(self, inputs, outputs, residuals):
+        """
+        Don't solve; just calculate the residual.
+        """
+
+        x = inputs['x']
+        y = outputs['y']
+        z = outputs['z']
+
+        residuals['y'] = y - x - 2.0*z
+        residuals['z'] = x*z + z - 4.0
+
+        self.counter += 1
+        if self.counter > 20 and self.counter < 25:
+            raise om.AnalysisError('catch me')
+
```

Comment:
Do you get any failures if you change the lower and upper checks here in the original `ImplCompTwoStatesAE` class? I think it is only used in 2 tests, and the other test deals with the iprint output.

If you do, then it's okay, but you can simplify this by just deriving a new class from `ImplCompTwoStatesGuess` with a replacement of the apply_nonlinear method.  A lot less code and one less unused class.